hmpps-kotlin-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/kotlin/auth/HmppsWebClientConfiguration.kt

```diff
@@ -115,27 +116,36 @@ class HmppsReactiveWebClientConfiguration {
 }
 
 fun oAuth2AuthorizedClientProvider(clientCredentialsClientTimeout: Duration): OAuth2AuthorizedClientProvider {
-  // TODO We have to use a deprecated class here because that's what Spring Security uses. If Spring Security switches to RestClientClientCredentialsTokenResponseClient then we can do the same.
-  val accessTokenResponseClient = DefaultClientCredentialsTokenResponseClient().kotlinApply {
-    setRestOperations(
-      RestTemplate(listOf(FormHttpMessageConverter(), OAuth2AccessTokenResponseHttpMessageConverter())).kotlinApply {
-        errorHandler = OAuth2ErrorResponseErrorHandler()
-        requestFactory = SimpleClientHttpRequestFactory().kotlinApply {
-          setReadTimeout(clientCredentialsClientTimeout)
-        }
-      },
-    )
+  val accessTokenResponseClient = RestClientClientCredentialsTokenResponseClient().kotlinApply {
+    val requestFactory = ReactorClientHttpRequestFactory().kotlinApply {
```

Comment:
I've switched over to a netty based client, based on the new default in https://github.com/spring-projects/spring-framework/blob/main/spring-web/src/main/java/org/springframework/web/client/DefaultRestClientBuilder.java#L458-L460